### PR TITLE
product card

### PR DESCRIPTION
**Pull request by AlShramchemko**
- [DEMO LINK](https://AlShramchemko.github.io/layout_product-cards/)
- [TEST REPORT LINK](https://AlShramchemko.github.io/layout_product-cards/report/html_report/)
-1
-2 созданы переменные, указанный margin занесен в общий, background: no-repeat занесен в  общий
-3 надеюсь ничего не пропустил
-4 переиминовал папку stars убрал box-sizing
-5 картинку както пропустил а за box-sizing не правильно понял

### CODE REVIEW

File: src/styles/index.scss [hunk 1]
```diff
@@ -1,3 +1,175 @@
...
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  width: 198px;
+  background-color: rgba(255, 255, 255, 1);
+  border: 1px solid #f3f3f3;
+  border-radius: 5px;
+
+  &:hover {
+    border: 1px solid #f3f3f3;
```
Comment (loralevitska) [inline src/styles/index.scss:41]: Repeatable colour move to variables

File: src/styles/index.scss [hunk 2]
```diff
@@ -1,3 +1,175 @@
...
+  &__image {
+    margin: 0;
+    padding: 0;
+    margin-top: 32px;
+    display: block;
+    width: 160px;
+    height: 134px;
+    background: url('../images/imac.jpeg');
+    background-size: 160px 134px;
+    background-repeat: no-repeat;
```
Comment (loralevitska) [inline src/styles/index.scss:53]: Combine  with `background` shorthand property

File: src/styles/index.scss [hunk 3]
```diff
@@ -1,3 +1,175 @@
...
+    font-family: Roboto-medium, sans-serif;
+    font-weight: 500;
+    line-height: 18px;
+    font-size: 12px;
+    color: #060b35;
+  }
+
+  &__productCode {
+    margin: 0;
+    margin-top: 4px;
```
Comment (loralevitska) [inline src/styles/index.scss:74]: Combine with `margin` shorthand property

**Review by loralevitska — CHANGES_REQUESTED** [review]
Please redeploy TEST REPORT LINK for review

File: src/index.html [hunk 4]
```diff
@@ -7,12 +7,62 @@
...
       href="./styles/index.scss"
     />
   </head>
   <body>
-    <h1>Product cards</h1>
+    <div
+      class="card"
+      data-qa="card"
+    >
+      <div class="card__image"></div>
```
Comment (volodymyr-soltys97) [inline src/index.html:25]: This image isn't background, so you need to add image with the img tag

File: src/index.html [hunk 5]
```diff
@@ -7,12 +7,62 @@
...
+
+      <div class="card__profuctName">
```
Comment (volodymyr-soltys97) [inline src/index.html:27]: This div is redundant
```suggestion
```

File: src/index.html [hunk 6]
```diff
@@ -7,12 +7,62 @@
...
+        <p>APPLE A1419 iMac 27" Retina 5K Monoblock (MNED2UA/A)</p>
+      </div>
+
+      <div class="card__productCode">
```
Comment (volodymyr-soltys97) [inline src/index.html:31]: The same
```suggestion
```

File: src/index.html [hunk 7]
```diff
@@ -7,12 +7,62 @@
...
+      </div>
+
+      <div class="card__stars">
+        <div class="card__checked">
+          <div class="card__star"></div>
+          <div class="card__star"></div>
+          <div class="card__star"></div>
+          <div class="card__star"></div>
+          <div class="card__star"></div>
+        </div>
```
Comment (volodymyr-soltys97) [inline src/index.html:42]: Need to use stars block from the previous task and add the modifier class
```suggestion
        <div class="stars stars--4">
          <div class="stars__star"></div>
          <div class="stars__star"></div>
          <div class="stars__star"></div>
          <div class="stars__star"></div>
          <div class="stars__star"></div>
        </div>
```

File: src/index.html [hunk 8]
```diff
@@ -7,12 +7,62 @@
...
+
+        <div class="card__reviews">
```
Comment (volodymyr-soltys97) [inline src/index.html:44]: Redundant
```suggestion
```

File: src/index.html [hunk 9]
```diff
@@ -7,12 +7,62 @@
...
+        method="post"
+        data-qa="hover"
+      >
+        <input
+          type="submit"
+          value="BUY"
+          class="card__buy"
+          data-qa="hover"
+        />
+      </form>
```
Comment (volodymyr-soltys97) [inline src/index.html:65]: Use the tag a for this button instead of this

File: src/styles/index.scss [hunk 10]
```diff
@@ -1,3 +1,174 @@
...
+
+@import '../utils/mixins';
+
+html {
+  font-family: Roboto, sans-serif;
+}
+
 body {
   margin: 0;
+  padding: 0;
```
Comment (volodymyr-soltys97) [inline src/styles/index.scss:29]: The body doesn't have padding
```suggestion
```

File: src/styles/index.scss [hunk 11]
```diff
@@ -1,3 +1,174 @@
...
+  width: 200px;
```
Comment (volodymyr-soltys97) [inline src/styles/index.scss:30]:
```suggestion
```

File: src/styles/index.scss [hunk 12]
```diff
@@ -1,3 +1,174 @@
...
+}
+
+.card {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  width: 198px;
```
Comment (volodymyr-soltys97) [inline src/styles/index.scss:37]:
```suggestion
  width: 200px;
```

File: src/styles/index.scss [hunk 13]
```diff
@@ -1,3 +1,174 @@
...
+    background-position: center;
+
+    &:not(:last-child) {
+      margin-right: 4px;
+    }
+
+    &:nth-child(-n + 4) {
+      background-image: url(../images/star-active.svg);
+      background-position: center;
+    }
```
Comment (volodymyr-soltys97) [inline src/styles/index.scss:112]: Move stars styles to the separate file

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
Good job 👍 
Let's improve your code
This part of the task is incomplete
image

File: src/src_stars/style.css
Comment (SanyaBratashchuk) [inline src/src_stars/style.css:1]: rename folder src_stars => stars

File: src/styles/index.scss [hunk 14]
```diff
@@ -1,3 +1,152 @@
...
+
+.card {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  width: 200px;
+  background-color: $button_color_hover;
+  border: 1px solid $border_color;
+  border-radius: 5px;
+  box-sizing: border-box;
```
Comment (SanyaBratashchuk) [inline src/styles/index.scss:39]:
```suggestion
```
It is common practice to apply `box-sizing: border-box;` for all elements via `*` selector

**Review by SanyaBratashchuk — CHANGES_REQUESTED** [review]
move each BEM block to a separate file

File: src/styles/index.scss [hunk 15]
```diff
@@ -1,3 +1,153 @@
...
+.card {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  width: 198px;
+  background-color: $button_color_hover;
+  border: 1px solid $border_color;
+  border-radius: 5px;
+
+  // box-sizing: border-box;
```
Comment (vadiimvooo) [inline src/styles/index.scss:40]: You haven't had to comment it. Just provide this style with `*` selector

File: src/styles/index.scss (hunk 12 above)
Comment (vadiimvooo) [reply src/styles/index.scss:35]: Agreed with Volodymyr

File: src/index.html (hunk 4 above)
Comment (vadiimvooo) [reply src/index.html:25]: Agreed with Volodymyr

**Review by vadiimvooo — CHANGES_REQUESTED** [review]
Good job, but you have to fix all prev comments. Feel free to ask for help in fe_chat if you need it.

File: src/styles/index.scss (hunk 15 above)
Comment (AlShramchemko) [reply src/styles/index.scss:40]: было сказано избаудалить border-box;

**Review by nazarmatsevych — APPROVED** [review]
Well done 👍

**Pull request closed without merging**